File: springboot/src/main/java/com/freshmall/entity/OrderItem.java
package com.freshmall.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 订单明细表（商品名称/图片/价格为下单时快照）
 */
@Data
@Entity
@Table(name = "order_items")
public class OrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "order_id", nullable = false)
    private Integer order_id;

    @Column(name = "goods_id", nullable = false)
    private Integer goods_id;

    @Column(name = "goods_name", nullable = false, length = 100)
    private String goods_name;

    @Column(name = "goods_image", length = 255)
    private String goods_image = "";

    /** 下单单价（快照） */
    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price;

    @Column(nullable = false)
    private Integer quantity;

    /** 小计 = 单价 × 数量 */
    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal subtotal;

    @Column(name = "created_at", updatable = false)
    @CreationTimestamp
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    @UpdateTimestamp
    private LocalDateTime updatedAt;
}

File: server/models/order_item.js
/**
 * 订单明细模型 → order_items 表
 * 商品名称/图片/价格均为下单时的快照，商品变更不影响历史订单展示
 */
const { DataTypes } = require('sequelize')
const sequelize = require('../config/database')

const OrderItem = sequelize.define('OrderItem', {
  id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
  order_id: { type: DataTypes.INTEGER, allowNull: false, comment: '所属订单 id' },
  goods_id: { type: DataTypes.INTEGER, allowNull: false, comment: '商品 id' },
  goods_name: { type: DataTypes.STRING(100), allowNull: false, comment: '商品名称（快照）' },
  goods_image: { type: DataTypes.STRING(255), defaultValue: '', comment: '商品主图（快照）' },
  price: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: false,
    comment: '下单单价（快照）',
  },
  quantity: { type: DataTypes.INTEGER, allowNull: false, comment: '购买数量' },
  subtotal: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: false,
    comment: '小计 = 单价 × 数量',
  },
}, {
  tableName: 'order_items',
})

module.exports = OrderItem
